labencryptedstorage/src/main/java/mobi/lab/labencryptedstorage/LabEncryptedStorageManager.kt

```diff
@@ -170,80 +192,72 @@ public open class LabEncryptedStorageManager(
-         * If a storage is already decided and
-         * this is override then all the already written values will become unreadable!
-         * Used for fallback flow and remembering the selected storage.
-         * Default: [KeyValueStorageClearTextSharedPreferences].
-         *
-         * @param clearTextStorageOverride Implementation for the clear-text storage.
+         * @param blockedDeviceManufacturerAndModel List of "manufacturer1 model1","manufacturer2 model2"
          */
-        public fun overrideClearTextStorage(clearTextStorageOverride: KeyValueClearTextStorage): Builder =
-            apply { this.clearTextStorage = clearTextStorageOverride }
+        public fun encryptionBlocklist(vararg blockedDeviceManufacturerAndModel: String): Builder =
+            apply { this.encryptionBlocklist.addAll(blockedDeviceManufacturerAndModel) }
 
         /**
-         * Override implementation for the encrypted storage.
-         * WARNING: In 99% of cases you do not want to set this.
-         * If a storage is already decided and
-         * this is override then all the already written values will become unreadable!
-         * Used if allowed and storageCompatibilityTester shows the device supports it.
-         * Default: [KeyValueStorageEncryptedSharedPreferences].
+         * Set the preferred hardware key store based encryption element.
+         * Available options are [mobi.lab.labencryptedstorage.entity.EncryptionPreferredType.Tee] and
+         * [mobi.lab.labencryptedstorage.entity.EncryptionPreferredType.StrongBox].
+         * Default is the former.
          *
-         * @param encryptedStorageOverride Implementation for the encrypted  storage.
+         * @param encryptionPreferredType Preferred type
          */
-        public fun overrideEncryptedTextStorageImplementation(encryptedStorageOverride: KeyValueEncryptedStorage): Builder =
-            apply { this.encryptedStorage = encryptedStorageOverride }
+        public fun encryptionPreferredType(encryptionPreferredType: EncryptionPreferredType): Builder =
+            apply { this.encryptionPreferredType = encryptionPreferredType }
 
         /**
          * Implementation for tester to test if the encrypted storage works on this given device.
          * Default: [SimpleEncryptedStorageDeviceCompatibilityTester].
-         * @param storageCompatibilityTester Implementation for tester to test if the encrypted storage works on this given device
+         *
+         * @param encryptionDeviceCompatibilityTester Implementation for tester to test if the encrypted storage works on this given device
          */
-        public fun storageCompatibilityTester(storageCompatibilityTester: EncryptedStorageCompatibilityTester): Builder =
-            apply { this.storageCompatibilityTester = storageCompatibilityTester }
+        public fun encryptionDeviceCompatibilityTester(
+            encryptionDeviceCompatibilityTester: EncryptedStorageCompatibilityTester
```

Comment:
The function name was changed but the interface class still uses the old name with "storage" in it. Maybe makes sense to use the same name for the interface, too?

Reply:
Good point